web/src/components/PoiFilters.tsx

```diff
@@ -88,6 +93,15 @@ const PoiFilters = ({
 }: PoiFiltersProps): ReactElement => {
   const { t } = useTranslation('pois')
 
+  const handleFilterChange = (event: React.MouseEvent<HTMLElement>, newValue: string | null) => {
+    if (newValue === null) {
+      setSelectedPoiCategory(null)
+    } else {
+      const category = poiCategories.find(cat => cat.id.toString() === newValue)
+      setSelectedPoiCategory(category ?? null)
+    }
+  }
```

Comment:
🔧 This can be simplified with only one call to `setSelectedPoiCategory`, could you try to do that please?